<launch>
	

	<node type="semantic_segmentation.py" name="semantic_segmentation" pkg="fcn8_ros" respawn="true" output="screen">
		<param name="topicInName" value="/Logitech_webcam/image_raw" />
		<param name="objectType_unknown" value="True" />
		<param name="objectType_grass" value="True" />
		<param name="objectType_ground" value="True" />
		<param name="objectType_human" value="True" />
		<param name="objectType_shelterbelt" value="True" />
		<param name="objectType_vehicle" value="True" />
		<param name="objectType_water" value="True" />
		<!-- Select and object type (0-19 is YOLO darknet_ros, 31 is a pedestrian detector)  20. SS_unknown upto 30. SS_water  -->
		<param name="imgDimWidth" value="400" />
		<param name="imgDimHeight" value="300" />
		<param name="dirModelDescription" value="$(find fcn8_ros)/models/deploy.prototxt" />
		<param name="dirModelVaules" value="$(find fcn8_ros)/models/pascalcontext-fcn8s-heavy.caffemodel" />
		<param name="dirTestImage" value="$(find fcn8_ros)/src/Street2.jpg" />
		<param name="dirRemapping" value="$(find fcn8_ros)/remappingObjectTypes.mat" />
	</node>


	<node name="republish" pkg="image_transport" type="republish" args="compressed in:=/Logitech_webcam/image_raw _image_transport:=compressed raw out:=/Logitech_webcam/image_raw" respawn="false" output="screen">
	</node> 

	<node name="image_raw" pkg="image_view" type="image_view" respawn="false" output="screen">
		<remap from="image" to="/Logitech_webcam/image_raw"/>
		<param name="autosize" value="false" />
	</node>
	
	<node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
		<remap from="image" to="det/Logitech_webcam/semantic_segmentation/human"/>
		 <param name="autosize" value="false" />
	</node>
<!--
	<node type="image2ism.py" name="image2ism" pkg="image_inverse_sensor_model" respawn="true" output="screen">
		<param name="objectTypeInt" value="26" />
		<param name="imageWidth" value="400" />
		<param name="imageHeight" value="300" />
		<param name="cam_xTranslation" value="0.2038" /> 
		<param name="cam_yTranslation" value="0.255" />
		<param name="cam_zTranslation" value="2.056" />
		<param name="cam_pitch" value="0.1963" />  
		<param name="cam_yaw" value="0.0" />
		<param name="cam_FOV" value="0.68067" />
		<param name="grid_resolution" value="0.5" />
		<param name="grid_xSizeInM" value="-1.0" />
		<param name="grid_ySizeInM" value="-1.0" />
	</node>
-->
</launch>
